File: v1/Demo/demo_get_pos.py
import sys, os
sys.path.append(os.path.join(os.path.dirname(__file__), "../"))
from Hand_Sdk.HandSocket import *
from Hand_Sdk.Handapi import *


def main():
    ip_left = "192.168.137.39" # lefthand
    # ip_right = "192.168.137.19" # righthand
    api = API(ip_left)
    pos = api.get_pos()

    if len(pos) != 6:
        logger.print_trace_error("get position failed")
    else:
        logger.print_trace("get position success", pos)

if __name__ == "__main__":
    main()

File: v1/Demo/source/demo_get_pos.cpp
#include "../include/Handapi.h"

using namespace HandProtocol;

int main()
{
    Handapi api;
    api.init(LEFT_IP);
    api.calibration();
    usleep(1000000);


    vector<float> pos = api.get_pos();

    if (pos.size() != 6)
    {
        Logger::get_instance()->print_trace_error("get position failed\n");
        return ResultCode::FAIL;
    }
    
    std::cout<<"get position success : ";
    for (int i = 0; i < pos.size(); ++i) {
        std::cout << pos[i] << " ";
    }
    std::cout << std::endl;


    return 0;
}